### PR TITLE
[10.0] purchase_fop_shipping: fixes and improvements

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**
### I fixed:

- missing import of security group
- some pylint issues
- Fop is reached if invoice total amount is higher or equal to FOP amount
### I improved:

- add fop amount on purchase order form (below the total amount)
- fop amount can be company dependent

### CODE REVIEW

**Comment by Cedric-Pigeon** [comment]
@OCA/procurement-purchase-maintainers Could anyone review this PR?

**Review by RobinetDenisAcsone — APPROVED** [review]

File: purchase_fop_shipping/models/partner.py [hunk 1]
```diff
@@ -10,4 +10,5 @@ class ResPartner(models.Model):
     _inherit = 'res.partner'
 
     fop_shipping = fields.Float(
-        'FOP shipping', help='Min purchase order amount for FOP shipping',)
+        'FOP shipping', help='Min purchase order amount for FOP shipping',
+        company_dependent=True)
```
Comment (rousseldenis) [inline purchase_fop_shipping/models/partner.py]: @Cedric-Pigeon Shouldn't have migration script for the current company (will data be kept) ?

Comment (Cedric-Pigeon) [reply purchase_fop_shipping/models/partner.py]: @rousseldenis You are right. Data are lost by how can we know the related company in a migration script? Not sure the current one is the best option.

Comment (rousseldenis) [reply purchase_fop_shipping/models/partner.py]: So, a comment in readme for that version ?

Comment (Cedric-Pigeon) [reply purchase_fop_shipping/models/partner.py]: @rousseldenis Done

Comment (rousseldenis) [inline purchase_fop_shipping/models/partner.py]: @Cedric-Pigeon Apart of that, IMHO this field should use decimal precision (and a float compare in _compute_fop_shipping_reached)

Comment (Cedric-Pigeon) [reply purchase_fop_shipping/models/partner.py]: @rousseldenis Done

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by Cedric-Pigeon** [comment]
@mourad-ehm could you have a look please?

**Review by mourad-ehm — APPROVED** [review]
Code review.
Thank's for improvement

File: purchase_fop_shipping/__manifest__.py [hunk 2]
```diff
@@ -12,8 +12,10 @@
     'category': 'Purchase',
     'depends': ['purchase'],
     'website': 'http://www.akretion.com/',
```
Comment (nikul-serpentcs) [inline purchase_fop_shipping/__manifest__.py]: https://github.com/OCA/purchase-workflow

**Review by nikul-serpentcs — COMMENTED** [review]
Nitpicking

File: purchase_fop_shipping/__manifest__.py (hunk 2 above)
Comment (Cedric-Pigeon) [reply purchase_fop_shipping/__manifest__.py]: @nikul-serpentcs Done. thanks

**Comment by HviorForgeFlow** [comment]
@Cedric-Pigeon solve travis issues please

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi @Cedric-Pigeon. 
first thanks for all that improvments. 

Well, it's quite problematic to have an update that drop all the values, in a stable module, and it is against the OCA policy.
I think that an upgrade script is required, to recover fop value. (move data from partner table to ir.property table).
- backup data in pre-migration script
- recreate data in post-migration script

what do you think ?

**Comment by benwillig** [comment]
Hi @legalsylvain, I added the script

**Pull request closed without merging**